pkg/wallet/wallet_list_actions_test.go

```diff
@@ -118,57 +111,34 @@ func (s *WalletTestSuite) TestWalletListActions() {
 		defer cleanup()
 
 		// and:
-		args := fixtures.DefaultWalletListActionsArgsWithIncludes()
+		createArgs := fixtures.DefaultWalletCreateActionArgs(t)
+		createResult, err := aliceWallet.CreateAction(t.Context(), createArgs, fixtures.DefaultOriginator)
+		require.NoError(t, err)
+		require.NotNil(t, createResult)
 
 		// when:
-		result, err := aliceWallet.ListActions(t.Context(), args, fixtures.DefaultOriginator)
+		listArgs := fixtures.DefaultWalletListActionsArgsWithIncludes()
+		listArgs.Labels = createArgs.Labels
+		result, err := aliceWallet.ListActions(t.Context(), listArgs, fixtures.DefaultOriginator)
 
 		// then:
 		assert.NoError(t, err)
 		require.NotNil(t, result)
-		assert.GreaterOrEqual(t, int(result.TotalActions), 0, "Should have non-negative total actions")
-
-		// then: verify the included fields are properly populated
-		if len(result.Actions) > 0 {
-			action := result.Actions[0]
-			assert.NotNil(t, action.Labels, "Labels should be included")
-			assert.NotNil(t, action.Inputs, "Inputs should be included")
-			assert.NotNil(t, action.Outputs, "Outputs should be included")
-		}
-	})
-
-	s.Run("list actions with pagination", func() {
-		t := s.T()
-
-		// given:
-		given := testabilities.Given(t)
+		assert.Equal(t, uint32(1), result.TotalActions, "Should have one action")
+		require.Len(t, result.Actions, 1)
 
 		// and:
-		aliceWallet, cleanup := given.AliceWalletWithStorage(s.StorageType)
-		defer cleanup()
-
-		// and:
-		args := fixtures.DefaultWalletListActionsArgs()
-		args.Limit = 2
-		args.Offset = 0
-
-		// when:
-		result, err := aliceWallet.ListActions(t.Context(), args, fixtures.DefaultOriginator)
-
-		// then:
-		assert.NoError(t, err)
-		require.NotNil(t, result)
-		assert.GreaterOrEqual(t, int(result.TotalActions), 0, "Should have non-negative total actions")
-
-		// then: test second page
-		args.Offset = 2
-		result2, err := aliceWallet.ListActions(t.Context(), args, fixtures.DefaultOriginator)
-		assert.NoError(t, err)
-		require.NotNil(t, result2)
-		assert.Equal(t, result.TotalActions, result2.TotalActions, "Total actions should be consistent across pages")
+		action := result.Actions[0]
+		assert.NotEmpty(t, action.Txid, "Should have a transaction ID")
+		assert.Equal(t, createArgs.Description, action.Description)
+		assert.Equal(t, createArgs.Version, action.Version)
+		assert.Equal(t, createArgs.LockTime, action.LockTime)
+		assert.ElementsMatch(t, createArgs.Labels, action.Labels)
+		assert.NotNil(t, action.Inputs, "Inputs should be included")
+		assert.NotNil(t, action.Outputs, "Outputs should be included")
 	})
 
-	s.Run("list actions with label query mode all", func() {
+	s.Run("list actions after internalize action", func() {
```

Comment:
Same as for create action.

Add another `when:` `then` section to create action tests.